src/main/scala/com/microsoft/ml/spark/explainers/LassoRegression.scala

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.explainers
+import breeze.linalg.{norm, sum, DenseMatrix => BDM, DenseVector => BDV}
+import breeze.numerics.abs
+
+import scala.annotation.tailrec
+
+private[spark] case class CoordinateDescentLasso(alpha: Double, maxIterations: Int, tol: Double) {
+  require(maxIterations >= 1)
+  require(tol >= 0)
+
+  def softThresholdingOp(x: Double, lambda: Double): Double = {
+    if (x > 0 && lambda < math.abs(x))
+      x - lambda
+    else if (x < 0 && lambda < math.abs(x))
+      x + lambda
+    else
+      0d
+  }
+
+  private def fitIteration(x: BDM[Double], y: BDV[Double])(currBeta: BDV[Double]): BDV[Double] = {
+    val newBeta = currBeta.copy
```

Comment:
can't the data also be sparse, how do we handle sparse vectors?

Reply:
I don't know if there is an algorithm that optimizes for sparse vectors. For now, I changed the interface inside `RegressionBase` (that's the public facing API) to support the more general Vector type, and just convert the vector inputs to dense type, so at least functional-wise, we can handle sparse vectors.

Reply:
@memoryz yes, we should be able to support sparse vector, please see PR:
https://github.com/slundberg/shap/pull/200
lasso regression should support sparse data, eg see:
https://scikit-learn.org/stable/auto_examples/linear_model/plot_lasso_dense_vs_sparse_data.html

we can discuss this more and improve in separate PR if you want

Reply:
Agree. Let's improve that in a separate PR.